src/index.ts

```diff
@@ -1,21 +1,93 @@
+
+import {
+    INotebookTracker
+} from '@jupyterlab/notebook';
+
+import {
+  ContentsManager
+} from '@jupyterlab/services';
+
+import {
+  ICommandPalette
+} from '@jupyterlab/apputils';
+
 import {
   JupyterLab,
   JupyterLabPlugin
 } from '@jupyterlab/application';
 
+import {AllOfUsConfig} from './config';
+import {ConceptsWidget} from './concepts-widget';
+
 import '../style/index.css';
 
+// Activate the jupyterhub extension.
+function activateExtension(app: JupyterLab,
+                           notebooks: INotebookTracker,
+                           palette: ICommandPalette): void {
+    const tracker = notebooks;
+    let allOfUsConfig = new AllOfUsConfig();
+
+    // Create a single widget
+    let conceptsWidget: ConceptsWidget = new ConceptsWidget(allOfUsConfig);
+
+    // Add an application command
+    const conceptsCommand: string = 'allOfUs:concepts';
+    app.commands.addCommand(conceptsCommand, {
+      label: 'All of Us Concepts',
+      execute: () => {
+        if (!conceptsWidget.isAttached) {
+          // Attach the widget to the main area if it's not there
+          app.shell.addToMainArea(conceptsWidget, { mode: 'split-bottom' });
+        }
+        // Activate the widget
+        app.shell.activateById(conceptsWidget.id);
+      }
+    });
+
+
+    // Add the command to the palette.
+    palette.addItem({command: conceptsCommand, category: 'All Of Us'});
+
+    let initial_load = true;
+    tracker.currentChanged.connect(() =>
+    {
+      if (tracker.currentWidget) {
```

Comment:
Is this always going to be set? If not, does this mean we might render the widget without a workspace context?

Reply:
Yes, that's possible. You could for example close all your notebooks. In which case we probably will just use whatever your last notebook was. 

If the widgets loads when no notebook has ever been loaded, we'll probably disable stuff within it / display a message saying "open a notebook".